### PR TITLE
End to end tests

### PR DESCRIPTION
**Pull request by em-pe**
#### Description

Workflow that runs spaceflights scenario on our GCP infrastructure. Currently workflow is configured in a way, that runs only on `develop` branch, and proper PR's (not draft). We may review the costs of running tests later on. 

Follow up tasks cover: 
- waiting for pipeline completion and checking the result #15 
- validating kedro-mlflow compatibility against our mlflow instance #17 

##### PR Checklist
- [x] Tests added
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        pip install . 'kedro<0.18'
+        kedro new --starter spaceflights --config tests/e2e/starter-config.yml --verbose --checkout=0.17.6
+
+    - name: Install project dependencies
+      run: |
+        cd ./spaceflights
+        echo "git+https://github.com/getindata/kedro-vertexai.git@$GITHUB_SHA" >> src/requirements.txt
+        echo "kedro-docker" >> src/requirements.txt
+        sed -i '/kedro-telemetry/d' src/requirements.txt
+        cat src/requirements.txt
```
Comment (szczeles) [inline .github/workflows/e2e-test.yml:39]: Leftover?

**Comment by szczeles** [comment]
Impressive!!!

**Review by szczeles — APPROVED** [review]

**Review by marrrcin — APPROVED** [review]
💪🏻

File: .github/workflows/e2e-test.yml (hunk 1 above)
Comment (em-pe) [reply .github/workflows/e2e-test.yml:39]: You mean this `cat`? I left it intentionally for debugging purposes

**Pull request merged**